solidity/test/VendingMachineTest.js

```diff
@@ -711,71 +712,66 @@ describe("VendingMachine", async function() {
-      let maxSupply = await vendingMachine.getMaxSupply()
-      expect(maxSupply).to.eq.BN(btcToTbtc(100))
-
-      await increaseTime(29 * 24 * 60 * 60 - 10 * 60) // 30 days minus 10 minutes
-      maxSupply = await vendingMachine.getMaxSupply()
-      expect(maxSupply).to.eq.BN(btcToTbtc(100))
-
-      await increaseTime(1 * 60 * 60) // one hour
-      maxSupply = await vendingMachine.getMaxSupply()
-      expect(maxSupply).to.not.eq.BN(btcToTbtc(100))
-    })
-
-    it("has a max supply of 250 BTC between the 30th day and 60th day", async () => {
-      await increaseTime(30 * 24 * 60 * 60 + 1) // 30 days and 1 second
-      let maxSupply = await vendingMachine.getMaxSupply()
-      expect(maxSupply).to.eq.BN(btcToTbtc(250))
-
-      await increaseTime(30 * 24 * 60 * 60 - 10 * 60) // 30 days minus 10 minutes
-      maxSupply = await vendingMachine.getMaxSupply()
-      expect(maxSupply).to.eq.BN(btcToTbtc(250))
-
-      await increaseTime(60 * 60) // one hour
-      maxSupply = await vendingMachine.getMaxSupply()
-      expect(maxSupply).to.not.eq.BN(btcToTbtc(250))
-    })
-
-    it("has a max supply of 500 BTC between the 60th day and 90th day", async () => {
-      await increaseTime(60 * 24 * 60 * 60 + 1) // 60 days and 1 second
-      let maxSupply = await vendingMachine.getMaxSupply()
-      expect(maxSupply).to.eq.BN(btcToTbtc(500))
-
-      await increaseTime(30 * 24 * 60 * 60 - 10 * 60) // 30 days minus 10 minutes
-      maxSupply = await vendingMachine.getMaxSupply()
-      expect(maxSupply).to.eq.BN(btcToTbtc(500))
-
-      await increaseTime(60 * 60) // one hour
-      maxSupply = await vendingMachine.getMaxSupply()
-      expect(maxSupply).to.not.eq.BN(btcToTbtc(500))
-    })
-
-    it("has a max supply of 21M BTC after the 90th day", async () => {
-      await increaseTime(90 * 24 * 60 * 60 + 1) // 120 days and 1 second
-      let maxSupply = await vendingMachine.getMaxSupply()
-      expect(maxSupply).to.eq.BN(btcToTbtc(21000000))
-
-      await increaseTime(30 * 24 * 60 * 60) // 30 days
-      maxSupply = await vendingMachine.getMaxSupply()
-      expect(maxSupply).to.eq.BN(btcToTbtc(21000000))
+    const MINUTE = 60
+    const HOUR = 60 * MINUTE
+    const DAY = 24 * HOUR
+
+    const SUPPLY_SCHEDULE = [
+      [0, 2 * DAY, 2],
+      [2 * DAY, 7 * DAY, 100],
+      [7 * DAY, 14 * DAY, 250],
+      [14 * DAY, 21 * DAY, 500],
+      [21 * DAY, 28 * DAY, 750],
+      [28 * DAY, 35 * DAY, 1000],
+      [35 * DAY, 42 * DAY, 1500],
+      [42 * DAY, 49 * DAY, 2000],
+      [49 * DAY, 56 * DAY, 2500],
+      [56 * DAY, 63 * DAY, 3000],
+      [63 * DAY, Infinity, 21000000],
+    ]
+
+    SUPPLY_SCHEDULE.forEach(([start, end, expectedMax]) => {
+      const humanize = function(interval) {
+        if (interval === 0) {
+          return "instantiation"
+        } else if (interval === Infinity) {
+          return "the end of time"
+        }
+        return moment.duration(interval, 'seconds').humanize({d: 7, w: 40})
```

Comment:
Well this was definitely overkill, particularly since it added a dependency (and its dependencies) to our install, but I like the spunk :grin:

Reply:
> since it added a dependency

You have no idea how long I agonized over that, lol. The closest existing dep I saw was `humanized` and it doesn't cover intervals... I could just write the code but look how cute

Reply:
Lolol. Mostly I think if the test output just said `0 days` or `Infinity days`, the person reading the test would understand well enough what it meant.

> look how cute

^^ yes, this :grin: